mobilecoind/src/bin/main.rs

```diff
@@ -159,26 +159,36 @@ fn create_or_open_ledger_db(
     logger: &Logger,
     transactions_fetcher: &ReqwestTransactionsFetcher,
 ) -> LedgerDB {
+    let ledger_db_file = Path::new(&config.ledger_db).join("data.mdb");
+
     // Attempt to open the ledger and see if it has anything in it.
-    if let Ok(ledger_db) = LedgerDB::open(&config.ledger_db) {
-        if let Ok(num_blocks) = ledger_db.num_blocks() {
-            if num_blocks > 0 {
-                // Successfully opened a ledger that has blocks in it.
-                log::info!(
-                    logger,
-                    "Ledger DB {:?} opened: num_blocks={} num_txos={}",
-                    config.ledger_db,
-                    num_blocks,
-                    ledger_db.num_txos().expect("Failed getting number of txos")
-                );
-                return ledger_db;
+    match LedgerDB::open(&config.ledger_db) {
+        Ok(ledger_db) => {
+            if let Ok(num_blocks) = ledger_db.num_blocks() {
+                if num_blocks > 0 {
+                    // Successfully opened a ledger that has blocks in it.
+                    log::info!(
+                        logger,
+                        "Ledger DB {:?} opened: num_blocks={} num_txos={}",
+                        config.ledger_db,
+                        num_blocks,
+                        ledger_db.num_txos().expect("Failed getting number of txos")
+                    );
+                    return ledger_db;
+                }
             }
         }
-    }
+        Err(err @ _) => {
+            // If the ledger database exists and we failed to open it, something is wrong
+            // with it and this requires manual intervention.
+            if ledger_db_file.exists() {
+                panic!("Failed to open ledger db {:?}: {:?}", config.ledger_db, err);
```

Comment:
Will the errors here translate into user instructions to try running the db-migrate script?

Reply:
This is what the error looks like: `Details: panicked at 'Failed to open ledger db "/home/eran/mobilecoin/sample_data/ledger": MetadataStore(VersionIncompatible(20200707, 20220222))', mobilecoind/src/bin/main.rs:184:17`
I can add a specific case for this particular error that points at the migration utility. I can also try embedding it into mobilecoind and adding a `--migrate` config flag that enables auto-migration

Reply:
Did both, now there's a descriptive error message as well as a way to run the migration straight from mobilecoind.
This will be useful for full-service as well, where I imagine we would just always want to run it on startup.